src/storage/heap_file.c

```diff
@@ -14539,6 +14541,17 @@ heap_dump_heap_file (THREAD_ENTRY * thread_p, FILE * fp, bool dump_records, cons
     }
 
   heap_dump (thread_p, fp, &hfid, dump_records);
+
+  error_code = heap_get_class_partitions (thread_p, &class_oid, &parts, &parts_count);
+  if (error_code != NO_ERROR)
+    {
+      return;
+    }
+
+  for (int i = 1; i < parts_count; i++)
+    {
+      heap_dump (thread_p, fp, &parts[i].class_hfid, dump_records);
+    }
```

Comment:
I think we need to free the allocated  memory
```suggestion
    }
    
    if (parts != NULL)
      {
        db_private_free (thread_p, parts);
      }
```

Reply:
There are `heap_clear_partition_info` function in `heap_file.c` which frees partitions info from `heap_get_class_partitions`. I think we can use it.

Reply:
Oh, then lets use that function.